back-end/helpers/user-helpers.js

```diff
@@ -18,6 +18,35 @@ async function getUserGitHubInfo(userID) {
   return userGitHubData;
 }
 
+async function getUserLanguage(userID) {
+  const query = 'SELECT language FROM github_info WHERE user_id = $1';
+  const userLanguage = await queryWithParameters(
+      query,
+      [userID]
+  );
+  return userLanguage.language;
+}
+
+async function getRecommendations(userID) {
+  const languageToMatch = await getUserLanguage(userID);
+
+  // user_id clause so they are not recommended to connect with themself.
+  // ordered randomly so that same recommendations don't happen every time.
```

Comment:
Is there a way to check that we haven't already recommended these users before (and they've been rejected as matches?)

Would we store these matches somewhere? or is the user supposed to see new matches every time he/she loads the page?

I sort of imagined a workflow where...the backend would store these recommendations in the connections table + assign them an ID. Then, they could be referenced during the next series of recommendations to be sure we are not repeating recommendations the user has already accepted or rejected. 

Maybe that's overkill for what we need to deliver here, though.

Reply:
Yeah, sorry I think I'm starting to understand that workflow better now than when I made this, that's making more sense. At first I thought it could just kind of be reworked with including a connection-id. Maybe the best way is to initialize with all possible matches and then just pull from there / adjust accept and reject accordingly? Is that the workflow you were imagining?

Reply:
@mcmoonstruck I was thinking for this sprint since connections were not in scope that we could use this query to show that recommended users are actually being sent to the frontend. Once we get the design down for connections, this query will most definitely be rewritten

@AndreEPaul

Reply:
Okay. That sounds good to me.